picard/ui/searchdialog.py

```diff
@@ -452,11 +457,258 @@ def load_selection(self, row):
             # No files associated. Just a normal search.
                 self.tagger.load_album(track["musicbrainz_albumid"])
         else:
-        # The track is a NAT
             if self.file_:
                 album = self.file_.parent.album
                 self.tagger.move_file_to_nat(track["musicbrainz_recordingid"])
                 if album._files == 0:
                     self.tagger.remove_album(album)
             else:
                 self.tagger.load_nat(track["musicbrainz_recordingid"], node)
+
+    def restore_state(self):
+        size = config.persist["tracksearchdialog_window_size"]
+        if size:
+            self.resize(size)
+        self.search_box.restore_checkbox_state()
+
+    def restore_table_header_state(self):
+        header = self.table.horizontalHeader()
+        state = config.persist["tracksearchdialog_header_state"]
+        if state:
+            header.restoreState(state)
+        header.setResizeMode(QtGui.QHeaderView.Interactive)
+
+    def save_state(self):
+        if self.table:
+            self.save_table_header_state()
+        config.persist["tracksearchdialog_window_size"] = self.size()
+
+    def save_table_header_state(self):
+        state = self.table.horizontalHeader().saveState()
+        config.persist["tracksearchdialog_header_state"] = state
+
+
+class AlbumSearchDialog(SearchDialog):
+
+    options = [
+        config.Option("persist", "albumsearchdialog_window_size", QtCore.QSize(720, 360)),
+        config.Option("persist", "albumsearchdialog_header_state", QtCore.QByteArray())
+    ]
+
+    def __init__(self, parent):
+        super(AlbumSearchDialog, self).__init__(parent)
+        self.cluster = None
+        self.setWindowTitle(_("Album Search Results"))
+        self.table_headers = [
+                _("Name"),
+                _("Artist"),
+                _("Format"),
+                _("Tracks"),
+                _("Date"),
+                _("Country"),
+                _("Labels"),
+                _("Catalog #s"),
+                _("Barcode"),
+                _("Language"),
+                _("Type"),
+                _("Status"),
+                _("Cover")
+        ]
+
+    _coverart_column = 12
```

Comment:
You should initialize this one using length of the table_headers list, in __init__() or even drop it and use the length of the list directly since it is used only twice.